app/src/main/java/com/cmput301f20t21/bookfriends/ui/request/RequestViewModel.java

```diff
@@ -0,0 +1,146 @@
+package com.cmput301f20t21.bookfriends.ui.request;
+
+import android.net.Uri;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+import com.cmput301f20t21.bookfriends.entities.Book;
+import com.cmput301f20t21.bookfriends.entities.Request;
+import com.cmput301f20t21.bookfriends.enums.BOOK_STATUS;
+import com.cmput301f20t21.bookfriends.enums.REQUEST_STATUS;
+import com.cmput301f20t21.bookfriends.services.BookService;
+import com.cmput301f20t21.bookfriends.services.RequestService;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.DocumentSnapshot;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class RequestViewModel extends ViewModel {
+    private MutableLiveData<Book> book;
+    private MutableLiveData<Uri> imageUri;
+    private MutableLiveData<List<Request>> requesters;
+
+    private final RequestService requestService = RequestService.getInstance();
+    private final BookService bookService = BookService.getInstance();
+
+    /**
+     * Function to get the book information from FireStore
+     * @param bookId we will query the book information based on the bookID
+     */
+    public void getBookInfo(String bookId) {
+        bookService.getBookById(bookId).addOnCompleteListener(task -> {
+            if (task.isSuccessful()) {
+                DocumentSnapshot document = task.getResult();
+                if (document.exists()) {
+                    String author = (String) document.get("author");
+                    String description = (String) document.get("description");
+                    String imageName = (String) document.get("imageName");
+                    String isbn = (String) document.get("isbn");
+                    String owner = (String) document.get("owner");
+                    String status = (String) document.get("status");
+                    String title = (String) document.get("title");
```

Comment:
there is actually a helper function called `getBookFromDocument()` in BookService that you can use

Reply:
Nice, I didn't know that. Will replace it